spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java

```diff
@@ -56,18 +57,36 @@ public void init() {
 				.getPropertiesStrategy();
 		HystrixCommandExecutionHook commandExecutionHook = HystrixPlugins.getInstance()
 				.getCommandExecutionHook();
+		HystrixConcurrencyStrategy concurrencyStrategy = detectRegisteredConcurrencyStrategy();
 
 		HystrixPlugins.reset();
 
 		// Registers existing plugins excepts the Concurrent Strategy plugin.
 		HystrixPlugins.getInstance().registerConcurrencyStrategy(
-				new SecurityContextConcurrencyStrategy(existingConcurrencyStrategy));
+				new SecurityContextConcurrencyStrategy(concurrencyStrategy));
 		HystrixPlugins.getInstance().registerEventNotifier(eventNotifier);
 		HystrixPlugins.getInstance().registerMetricsPublisher(metricsPublisher);
 		HystrixPlugins.getInstance().registerPropertiesStrategy(propertiesStrategy);
 		HystrixPlugins.getInstance().registerCommandExecutionHook(commandExecutionHook);
 	}
 
+	private HystrixConcurrencyStrategy detectRegisteredConcurrencyStrategy() {
+		HystrixConcurrencyStrategy registeredStrategy = HystrixPlugins.getInstance()
+				.getConcurrencyStrategy();
+		if (existingConcurrencyStrategy == null) {
+			return registeredStrategy;
+		}
+		//Hystrix registered a default Strategy.
+		if (registeredStrategy instanceof HystrixConcurrencyStrategyDefault){
+			return existingConcurrencyStrategy;
+		}
+		//If registeredStrategy not the default and not some with existingConcurrencyStrategy throw a exception.
+		if (!existingConcurrencyStrategy.equals(registeredStrategy)){
```

Comment:
Why throw an exception in this case?  I think its fine to just return `existingConcurrencyStrategy`.

Reply:
if existingConcurrencyStrategy is not some with `registeredStrategy`, We can't simply use the `existingConcurrencyStrategy`, if we do that the  wrapper code logic in `registeredStrategy` will lost, obviously this will lead to a bug.
The exception for inform the user this case is not correct, and they must deal with case.
Do you have any other advice(inform the user this case is not correct, and they must deal with case)?

Reply:
Say that `existingConcurrencyStrategy` is not `null` (they provided a bean) and they also set a `registeredStrategy` why would returning `existingConcurrencyStrategy` be wrong?  Yes it doesnt make too much sense to have set both but I dont see how choosing `existingConcurrencyStrategy` as the default return value being a bug.